pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -77,12 +79,91 @@ func (r *AtlasDeploymentReconciler) ensureAdvancedDeploymentState(ctx *workflow.
 	}
 }
 
+func handleSearchNodes(ctx *workflow.Context, deployment *akov2.AtlasDeployment, projectID string) workflow.Result {
+	// Get the current state from Atlas
+	ctx.Log.Debug("starting search node processing")
+	defer ctx.Log.Debug("finished search node processing")
+
+	// NOTE: Despite being an array/slice, the Atlas API only allows 1 item.
+	nodesInAtlasEmpty := false
+	currentNodesInAtlas, httpResp, err := ctx.SdkClient.AtlasSearchApi.GetAtlasSearchDeployment(ctx.Context, projectID, deployment.GetDeploymentName()).Execute()
+	if err != nil {
+		if httpResp == nil {
+			ctx.Log.Debugf("unable to get current search nodes status: %v", err)
+			return workflow.Terminate(workflow.SearchNodesReady, err.Error())
+		}
+
+		switch httpResp.StatusCode {
+		// TODO: Should be only NotFound: CLOUDP-239015
+		case http.StatusBadRequest, http.StatusNotFound:
+			ctx.Log.Debug("no search nodes in atlas found")
+			nodesInAtlasEmpty = true
+		default:
+			return workflow.Terminate(workflow.SearchNodesReady, err.Error())
+		}
+	}
+
+	nodesInAkoEmpty := deployment.Spec.DeploymentSpec.SearchNodes == nil || len(deployment.Spec.DeploymentSpec.SearchNodes) == 0
+
+	switch {
+	case !nodesInAkoEmpty && nodesInAtlasEmpty:
+		// If no nodes configured in atlas, but some in the operator - create them.
+		ctx.Log.Debugf("creating search nodes %v", deployment.Spec.DeploymentSpec.SearchNodes)
+		_, _, err = ctx.SdkClient.AtlasSearchApi.CreateAtlasSearchDeployment(ctx.Context, projectID, deployment.GetDeploymentName(), &admin.ApiSearchDeploymentRequest{
+			Specs: deployment.Spec.DeploymentSpec.SearchNodesToAtlas(),
+		}).Execute()
+		if err != nil {
+			ctx.Log.Debugf("unable to create search nodes: %v", err)
```

Comment:
two questions:
1. Why is this a `ctx.Log.Debugf` and not a `ctx.Log.Errorf`?
2. Generally: Can't we handle non-`workflow.OK()` results including error logs higher in the chain to reduce chattiness here?

Reply:
agreed, with adding more meaningful reasons to the status conditions we can move logging higher in the chain.